tools/git-change-exec/actions-lint.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type lintSpdx struct {
+	extsMap    map[string]func(path string)
+	pathsToFix []string
+}
+
+func (s *lintSpdx) dontfix(path string) {
+	log.Printf("Cannot fix %s ...\n", path)
+}
+
+func (s *lintSpdx) yamlfix(path string) {
+	license := []string{
+		"# Copyright (c) 2024 Zededa, Inc.\n",
+		"# SPDX-License-Identifier: Apache-2.0\n\n",
+	}
+
+	log.Printf("Fixing %s ...\n", path)
+	prepend(path, license)
+}
+
+func (s *lintSpdx) gofix(path string) {
+	license := []string{
+		"// Copyright (c) 2024 Zededa, Inc.\n",
+		"// SPDX-License-Identifier: Apache-2.0\n\n",
+	}
+
+	log.Printf("Fixing %s ...\n", path)
+	prepend(path, license)
+}
+
+func (s *lintSpdx) dockerfilefix(path string) {
+	license := []string{
+		"# Copyright (c) 2024 Zededa, Inc.\n",
+		"# SPDX-License-Identifier: Apache-2.0\n\n",
+	}
+
+	log.Printf("Fixing %s ...\n", path)
+	prepend(path, license)
+}
+
+func prepend(path string, license []string) {
+	backupFh, err := os.CreateTemp("/var/tmp", "git-change-exec-spdx-fix")
+	if err != nil {
+		log.Fatalf("could not create temp file: %v", err)
+	}
+	backupPath := backupFh.Name()
+	defer os.Remove(backupPath)
+
+	for _, line := range license {
+		fmt.Fprint(backupFh, line)
+	}
+
+	origFh, err := os.Open(path)
+	if err != nil {
+		log.Fatalf("could not open original file %s: %v", path, err)
+	}
+
+	_, err = io.Copy(backupFh, origFh)
+	if err != nil {
+		log.Fatalf("could not copy: %v", err)
+	}
+
+	backupFh.Close()
+	origFh.Close()
+
+	err = copyFile(backupPath, path)
+	if err != nil {
+		log.Fatalf("could not rename %s -> %s: %v", backupPath, path, err)
+	}
+
+}
+
+func (s *lintSpdx) init() {
+	s.extsMap = map[string]func(path string){
+		".sh":        s.dontfix,
+		".go":        s.gofix,
+		".c":         s.dontfix,
+		".h":         s.dontfix,
+		".py":        s.dontfix,
+		".rs":        s.dontfix,
+		".yaml":      s.yamlfix,
+		".yml":       s.yamlfix,
+		"Dockerfile": s.dockerfilefix,
+	}
+
+	s.pathsToFix = make([]string, 0)
+}
+
+func (s *lintSpdx) hasSpdx(path string) bool {
```

Comment:
In this case, I would prefer just to run the already implemented script.
Otherwise, we must maintain the same logic in different places: the script in the tools and this one. Currently, the logic mentioned here differs from what is in the script.

Reply:
Now it is using the script.

Reply:
Do you think, I should directly call your script or keep this:
```
func (s *lintSpdx) hasSpdx(path string) bool {
        scriptPath := filepath.Join(filepath.Dir(s.ownPath), "..", "spdx-check.sh")

        cmd := exec.Command(scriptPath, path)
        err := cmd.Run()

        return err == nil
}

func (s *lintSpdx) match(path string) bool {
        if s.extsMap == nil {
                s.init()
        }

        if strings.Contains(path, "/vendor/") {
                return false
        }
        _, found := s.pathMatch(path)
        if !found {
                return false
        }
        if !s.hasSpdx(path) {
                s.pathsToFix = append(s.pathsToFix, path)
                return true
        }

        return false
}

```